### ISSUE TITLE
[a11y]: ComposedModal with scrolling content doesn't bring element into viewport with focus

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#19277]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### Automated testing tool and ruleset

N/A

### Description

When composed modal has scrolling content, moving the focus down the modal content using TAB only brings the _label_ of the focused element into view while leaving the element itself completely obscured. Since the element receiving focus is the input (not including the label) and is not obscured by a specific user action, I believe this is a violation of WCAG 2.4.11.

### WCAG 2.1 Violation

WCAG 2.4.11: Focus Not Obscured (Minimum)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-composedmodal--with-scrolling-content

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/?path=/story/components-composedmodal--with-scrolling-content
2. Adjust zoom to ~150% so that you must scroll to make some modal content visible
3. Move focus by pressing TAB until "Region" input is in focus
4. Press TAB once more. Observe that the "Dropdown" input is in focus but completely obscured

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
